rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,386 @@
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.torch.torch_rl_module import TorchRLModule
+from ray.rllib.core.rl_module.torch.tests.test_torch_rl_module import (
+    to_numpy,
+    to_tensor,
+)
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.models.specs.specs_dict import ModelSpec
+from ray.rllib.models.specs.specs_torch import TorchTensorSpec
+from ray.rllib.models.torch.torch_distributions import (
+    TorchDeterministic,
+    TorchCategorical,
+)
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.test_utils import check
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+from ray.rllib.utils.typing import TensorType
+
+
+def model_norm(model: torch.nn.Module) -> float:
+    """Compute the norm of the weights of a model.
+
+    Args:
+        model: The model whose weights to compute a norm over.
+
+    Returns:
+        the norm of model.
+    """
+    total_norm = 0
+    for p in model.parameters():
+        param_norm = p.detach().data.norm(2)
+        total_norm += param_norm.item() ** 2
+    total_norm = total_norm**0.5
+    return total_norm
+
+
+def do_rollouts(
+    env: gym.Env, module_for_inference: RLModule, num_rollouts: int
+) -> SampleBatch:
+    _returns = []
+    _rollouts = []
+    for _ in range(num_rollouts):
+        obs = env.reset()
+        _obs, _next_obs, _actions, _rewards, _dones = [], [], [], [], []
+        _return = -0
+        for _ in range(env._max_episode_steps):
+            _obs.append(obs)
+            fwd_out = module_for_inference.forward_inference(
+                {"obs": to_tensor(obs)[None]}
+            )
+            action = to_numpy(fwd_out["action_dist"].sample().squeeze(0))
+            next_obs, reward, done, _ = env.step(action)
+            _next_obs.append(next_obs)
+            _actions.append([action])
+            _rewards.append([reward])
+            _dones.append([done])
+            _return += reward
+            if done:
+                break
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+class BCTorchModule(TorchRLModule):
+    def __init__(self, config: FCConfig) -> None:
+        super().__init__(config)
+        self.policy = FCNet(config)
+
+    def input_specs(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+                "actions": TorchTensorSpec("b"),
+            }
+        )
```

Comment:
should we create a couple of util funcs on RLModule?
```
def default_input_specs(self, additional_columns=None):
    obs_dim = self.config.input_dim
    return ModelSpec(
        {
            "obs": TorchTensorSpec("b, do", do=obs_dim),
            **(additional_columns or {}),
        }
    )

def default_output_specs(self):
    return ModelSpec({"action_dist": Distribution})
```
then
```
def input_specs_exploration(self) -> ModelSpec:
    self.default_input_specs()

def input_specs_inference(self) -> ModelSpec:
    self.default_input_specs()

def input_specs_train(self) -> ModelSpec:
    self.default_input_specs({
        "actions": TorchTensorSpec("b"),
    })

def output_specs_train(self) -> ModelSpec:
    return self.default_output_specs()

def output_specs_inference(self) -> ModelSpec:
    return self.default_output_specs()

def output_specs_exploration(self) -> ModelSpec:
    return self.default_output_specs()
```

Reply:
Isn't this the same as having the user create a small helper method for getting the default specs and then calling that helper method from the main APIs? I feel like having an additional default method makes this less transparent and less explicit.

Reply:
yeah, you can argue about that.
my thinking was input specs with only obs column, and output spec with only action_dist column will be the dominating majority of cases.
maybe it's not worth it to ask users write this over and over again.